File: c/49.c
/* 296962999629 */
#include <stdio.h>
#include <stdlib.h>
#include "euler.h"

#define D 4
#define N 10000
#define S 3330

int main(int argc, char **argv)
{
    char *sieve;
    int a, aa, b, c;
    prime_sieve(N, &sieve, NULL);
    for (a = N/10+1; a < N; a += 2)
    {
        b = a + S;
        c = b + S;
        if (a != 1487 && sieve[a] && sieve[b] && sieve[c])
        {
            aa = digit_usage(a, 0);
            if (aa == digit_usage(b, 0) && aa == digit_usage(c, 0))
            {
                printf("%d%d%d\n", a, b, c);
                free(sieve);
                return 0;
            }
        }
    }
    return 0;
}

File: py/49.py
# 296962999629
import euler
D = 4
N = 10 ** D
S = 3330
sieve = euler.prime_sieve(N)
for a in xrange(N/10+1, N, 2):
    b = a + S
    c = b + S
    if a != 1487 and sieve[a] and sieve[b] and sieve[c]:
        aa = euler.digit_usage(a)
        if aa == euler.digit_usage(b) and aa == euler.digit_usage(c):
            print '%d%d%d' % (a, b, c)
            break
